portal/models/organization.py

```diff
@@ -855,20 +855,26 @@ def at_and_above_ids(self, organization_id):
             node = node.parent
         return ids
 
-    def find_top_level_org(self, organizations):
-        """Returns top level organization(s) based on the organizations provided
+    def find_top_level_orgs(self, organizations, first=False):
+        """Returns top level organization(s) from those provided
 
-        :param organizations: organizations against which top level organization(s) will be queried
+        :param organizations: organizations against which top level
+         organization(s) will be queried
+        :param first: if set, return the first org in the result list
+         rather than a set of orgs.
 
-        :return: list of top level organization(s)
+        :return: set of top level organization(s), or a single org if
+         ``first`` is set.
 
         """
-        orgs_list = []
+        results = set()
         for org in (o for o in organizations if o.id):
             top_org_id = self.find(org.id).top_level()
-            orgs_list.append(Organization.query.get(top_org_id))
+            results.add(Organization.query.get(top_org_id))
 
-        return orgs_list
+        if first:
+            return next(iter(results)) if results else None
```

Comment:
I don't think this will necessarily return the first org by id, sets don't retain order. Not sure if that's an issue however.

Reply:
Why not call `next(iter(results))` on the return value of `find_top_level_orgs()`, after the function call?

Reply:
Correct - but at this time, we are forced to not care.  I have the one log message to help where it matters.

I noticed *most* usage was just indexing the first element - also not certain to return a "first" - included this code in the common function to make for clean/clear client usage.

Reply:
Makes sense, maybe use an `OrderedDict` to mimic the old behavior (ordering), unless you're certain the ordering doesn't matter.

Reply:
I'm certain that it was indeterminate before, and no worse now.
You are assuming we know *which* org we might want, and that therefore ordering would help.  A false assumption.  The real fix is two fold:
1. hard business rule enforcement (we have a number of checks, leading us to primarily holding this true via annoying emails we fix).
2. a refactor to the QB engine so we can obtain the correct QB -> Org association at a low level.

I'd like to push this out as is.  It certainly doesn't make it any worse, and will eliminate at least one known false positive.